net/kcptun/Makefile

```diff
@@ -1,74 +1,65 @@
+PKG_LICENSE_FILES:=LICENSE.md
 
 PKG_BUILD_DEPENDS:=golang/host
 PKG_BUILD_PARALLEL:=1
 PKG_USE_MIPS16:=0
 
 GO_PKG:=github.com/xtaci/kcptun
 
-include $(INCLUDE_DIR)/package.mk
-include ../../lang/golang/golang-package.mk
-
-define Package/kcptun/template
-  SECTION:=net
-  CATEGORY:=Network
-  SUBMENU:=Web Servers/Proxies
-  TITLE:=KCP-based Secure Tunnel 
-  URL:=https://github.com/xtaci/kcptun
-  DEPENDS:=$(GO_ARCH_DEPENDS)
-endef
-
-define Package/kcptun-c
-  $(call Package/kcptun/template)
-  TITLE+= (client)
-endef
-
-define Package/kcptun-s
-  $(call Package/kcptun/template)
-  TITLE+= (server)
-endef
-
-define Package/kcptun/description
-  kcptun is a Stable & Secure Tunnel Based On KCP with N:M Multiplexing
-endef
-Package/kcptun-c/description = $(Package/kcptun/description)
-Package/kcptun-s/description = $(Package/kcptun/description)
-
-GO_PKG_LDFLAGS_X:=main.VERSION=$(PKG_VERSION)
 GO_PKG_LDFLAGS:=-s -w
+GO_PKG_LDFLAGS_X:=main.VERSION=$(PKG_VERSION)
 
-define Package/kcptun/install
-	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
-
-	$(INSTALL_DIR) $(1)/usr/bin/
-	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/$(2) $(1)/usr/bin/$(3)
-	$(INSTALL_DIR) $(1)/etc/config/
-	$(INSTALL_CONF) ./files/$(3).conf $(1)/etc/config/$(3)
-	$(INSTALL_DIR) $(1)/etc/init.d/
-	$(INSTALL_BIN) ./files/$(3).init $(1)/etc/init.d/$(3)
-endef
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
 
-define Package/kcptun-c/install
-  $(call Package/kcptun/install,$(1),client,kcptun-c)
+define Package/kcptun/Default
+  define Package/kcptun-$(1)
+    SECTION:=net
+    CATEGORY:=Network
+    SUBMENU:=Web Servers/Proxies
+    TITLE:=KCP-based Secure Tunnel $(1)
+    URL:=https://github.com/xtaci/kcptun
+    DEPENDS:=$$(GO_ARCH_DEPENDS)
+  endef
+
+  define Package/kcptun-$(1)/description
+    kcptun is a Stable & Secure Tunnel Based On KCP with N:M Multiplexing.
+This package only contains kcptun $(1).
+  endef
+
+  define Package/kcptun-$(1)/install
+		$$(call GoPackage/Package/Install/Bin,$$(PKG_INSTALL_DIR))
+
+		$$(INSTALL_DIR) $$(1)/usr/bin
+		$$(INSTALL_BIN) $$(PKG_INSTALL_DIR)/usr/bin/$(1) $$(1)/usr/bin/kcptun-$(1)
+		$$(INSTALL_DIR) $$(1)/etc/config
+		$$(INSTALL_CONF) ./files/kcptun.config $$(1)/etc/config/kcptun
+		$$(INSTALL_DIR) $$(1)/etc/init.d
+		$$(INSTALL_BIN) ./files/kcptun.init $$(1)/etc/init.d/kcptun
+  endef
 endef
 
-define Package/kcptun-s/install
-  $(call Package/kcptun/install,$(1),server,kcptun-s)
+KCPTUN_COMPONENTS:=server client
+define kcptun/templates
+  $(foreach component,$(KCPTUN_COMPONENTS),
+    $(call Package/kcptun/Default,$(component))
+  )
 endef
+$(eval $(call kcptun/templates))
 
-$(eval $(call GoBinPackage,kcptun-c))
-$(eval $(call BuildPackage,kcptun-c))
-$(eval $(call GoBinPackage,kcptun-s))
-$(eval $(call BuildPackage,kcptun-s))
+$(foreach component,$(KCPTUN_COMPONENTS), \
+  $(eval $(call GoBinPackage,kcptun-$(component))) \
+  $(eval $(call BuildPackage,kcptun-$(component))) \
+)
```

Comment:
Why not include `$(call Package/kcptun/Default,$(component))` from `kcptun/templates` in this loop?

Reply:
@jefferyto Fixed, thanks for your suggestion!